src/utils/hooks/useDataViews.tsx

```diff
@@ -168,34 +174,33 @@ const buildFilterDefinitions = (tableData?: DataViewsTableResponse): FilterDefin
       },
     }
 
-  headerRow.forEach(({ dataType, columnName, title, sortColumn, isBasicField }) => {
-    switch (dataType) {
-      case 'Date':
-        filterDefinitions[columnName] = {
-          type: 'date',
-          default: false,
-          visibleTo: [],
-          title,
-          options: {},
-        }
-        break
-      case 'string':
-        filterDefinitions[columnName] = {
-          type: 'searchableListIn',
-          default: false,
-          visibleTo: [],
-          title,
-          options: {
-            getFilterList: async () => {
-              return await getRequest(
-                getServerUrl('dataViews', { dataViewCode: code, column: columnName })
-              )
-            },
-          },
-        }
-        break
+  filterDefinitions.forEach(
+    ({ column, title, dataType, showFilterList, searchFields, delimiter, booleanMapping }) => {
+      returnFilterDefinitions[column] = {
+        type: filterTypeMap[dataType],
+        default: false,
+        visibleTo: [],
+        title,
+        options: {
+          column,
+          code,
+          dataType,
+          showFilterList,
+          searchFields,
+          delimiter,
+          booleanMapping,
+          nullString,
+        },
+      }
     }
-  })
+  )
+
+  return returnFilterDefinitions
+}
 
-  return filterDefinitions
+const filterTypeMap: { [key: string]: FilterTypes } = {
+  string: 'dataViewString',
+  number: 'dataViewNumber',
+  boolean: 'dataViewBoolean',
+  Date: 'date',
```

Comment:
In `<ListFilters/>`, the values "string", "boolean" etc. are already mapped to ApplicationList filters. So I'm remapping the data types to new keys here so we can map them to different ones. (Note that "date" is still using the ApplicationList date filter unmodified)